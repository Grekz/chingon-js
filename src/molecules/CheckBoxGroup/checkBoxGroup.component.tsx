import styled from '@emotion/styled'
import * as React from 'react'
import CheckBox from '../../atoms/CheckBox'

interface ICheckBoxGroupProps {
  options?: string[]
}

type CheckBoxGroupType = React.SFC<ICheckBoxGroupProps>

const CheckBoxGroup = (props: ICheckBoxGroupProps) => {
  const { options } = props
  return options ? (
    <CheckBoxGroupWrapper>
      {options.map((item, index) => (
        <CheckBox id={`checkBox_${index}`} key={index}>
          {item}
        </CheckBox>
      ))}
    </CheckBoxGroupWrapper>
  ) : null
}

const CheckBoxGroupWrapper: CheckBoxGroupType = styled.div``

export default CheckBoxGroup
