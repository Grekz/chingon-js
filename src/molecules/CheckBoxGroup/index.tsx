export { default } from './checkBoxGroup.component'
