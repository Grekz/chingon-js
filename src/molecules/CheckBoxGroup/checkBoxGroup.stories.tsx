import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import CheckBoxGroup from '.'

storiesOf('Molecules|CheckBoxGroup', module).add('As Default', () => (
  <CheckBoxGroup options={['Hello', 'World']} />
))
