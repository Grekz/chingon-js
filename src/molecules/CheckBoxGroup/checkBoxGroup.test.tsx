import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import CheckBoxGroup from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<CheckBoxGroup /> rendering', () => {
  it('should render CheckBoxGroup', () => {
    const wrapper = shallow(<CheckBoxGroup options={['Hello', 'World']} />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<CheckBoxGroup options={['Hello', 'World']} />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
