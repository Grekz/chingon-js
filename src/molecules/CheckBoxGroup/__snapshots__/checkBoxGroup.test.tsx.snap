// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBoxGroup /> rendering should match snapshot 1`] = `
<Styled(div)>
  <CheckBox
    id="checkBox_0"
    key="0"
  >
    Hello
  </CheckBox>
  <CheckBox
    id="checkBox_1"
    key="1"
  >
    World
  </CheckBox>
</Styled(div)>
`;
